cmd/deploy.go

```diff
@@ -302,18 +348,38 @@ func runDeploy(cmd *cobra.Command, newClient ClientFactory) (err error) {
 // optional build step.
 func build(cmd *cobra.Command, flag string, f fn.Function, client *fn.Client, buildOptions []fn.BuildOption) (fn.Function, error) {
 	var err error
+
+	// TODO: gauron99 - This checks whether a registry has changed on a subsequent build
+	// decision (Like on func deploy --registry=X when already built image is Y)
+	// This can probably be added into f.Built() function for simpler and more global
+	// implementation
+	registryChangedSubsequently := false
+	if !strings.Contains(f.Build.Image, f.Registry) && f.Image == "" && f.Build.Image != "" {
+		registryChangedSubsequently = true
+	}
+
+	build, _ := strconv.ParseBool(flag)
```

Comment:
I agree with your comment here.

Let's see if I can remember/understand the logic:   We want to rebuild if the user has changed the desired registry.  Another way to think of it being:  If the function's metadata indicates that the _next_ build of the function would result in a different image than is currently listed as the last one built (f.Build.Image), then it must be because the user has changed either f.Registry.

So yes, I think this could probably go in f.Built().  Doing so might also allow dropping the `&& f.Image == "" && f.Build.Image != ""` additional conditions:

The conditional `&& f.Image == ""` I think may not need to be included, because we want to treat the function as unbuilt if the registry changed since last build, even if no explicit image was provided as an override (the purpose of `f.Image`).

The subsequent conditional `&& f.Build.Image != ""` may also not need to be included, because the function should be considered unbuilt if there has been no previous build ever, which is covered by the primary case `!strings.Contains(f.Build.Image, f.Registry)`....  That will return true (function unbuilt), if `f.Build.Image` is empty (never been built before).

Perhaps try, and see if the tests pass :)